### DISCUSSION TITLE
Struggling to deploy a simple starter template hono and nodejs on vercel

### DISCUSSION THREAD

**Discussion by ghost** [honojs/hono#3471] [Q&A]
As you can read in the title, that's the problem i'm facing now, it's been 4 hours now looking for a solution for this but can't solve it. i'm trying to deploy a simple hono/nodejs on vercel but it keep saying 404 not found

```ts
// src/index.ts
import { Hono } from "hono";
import { serve } from "@hono/node-server";
import { handle } from "@hono/node-server/vercel"; 
import { logger } from "hono/logger";
import { cors } from "hono/cors";

const app = new Hono();

app.use(logger());
app.use(cors());

app.get("/", (c) => {
    return c.text("Hey There!")
});

serve({ fetch: app.fetch, port: 3000 }).on("listening", () => console.log("Listening..."));

export default handle(app);
```

```json
// package.json
{
    "name": "hono-app",
    "type": "module",
    "scripts": {
        "dev": "tsx watch src/index.ts"
    },
    "dependencies": {
        "@hono/node-server": "^1.13.1",
        "hono": "^4.6.3"
    },
    "devDependencies": {
        "@types/node": "^20.11.17",
        "tsx": "^4.7.1"
    }
}
```

**Comment by goMagFit** [comment]
app.options('/', (c) => {
  c.res.headers.append('Access-Control-Allow-Origin', '*');
  c.res.headers.append('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  c.res.headers.append('Access-Control-Allow-Headers', 'Content-Type');
  return c.text('OK', 200);
}); try adding this and then get to the root of your backend and run npm run dev and that should start up the server locally

**Reply by ghost** [reply]
```ts
app.use(cors());
```

This should do the work.

My problem is not with running the dev server locally, i want to deploy it to vercel, locally it's working just fine, i know it's something with `vercel.json` config but i'm not sure about it

**Comment by ghost** [comment]
i've solved this issue, found [this](https://github.com/pkorac/serverlesshono/tree/main) and it's working just fine
